### DISCUSSION TITLE
[Linux] How to add a .desktop file for creating shortcut

### DISCUSSION THREAD

**Discussion by unreal4u** [f1multiviewer/issue-tracker#313] [Show and tell]
I've created a small .desktop file so that I can search for the application and also find it in my settings page.

The file is called `multiviewer.desktop` and resides in `~/.local/share/applications/`. The contents of the file are: 

```
[Desktop Entry]
Version=1.0
Type=Application
Name=Multiviewer for F1
Comment=Experience F1TV® like never before
Exec='/home/unreal4u/f1/MultiViewer for F1-linux-x64/MultiViewer for F1' # <- enter absolute path here
Icon=/home/unreal4u/f1/MultiViewer for F1-linux-x64/resources/app/.webpack/main/88a36af69fdc182ce561a66de78de7b1.png # <- enter absolute path here
Path=/home/unreal4u/f1/MultiViewer for F1-linux-x64 # <- enter absolute path here
Categories=Internet;Multimedia
Terminal=false
StartupNotify=false
```

Now opening the application finder or looking for the application in either the Internet or Multimedia categories it will look like this:

![Screenshot_2022-11-22_11-31-38](https://user-images.githubusercontent.com/194999/203291888-6ba92ee9-2bc0-4154-aa0b-1e4029e10964.png)

(For the application to appear in either the Internet or Multimedia category you'll have to log out and back in).

I believe GNOME and KDE also support this format, but I'm not sure since I've not used those in years. Either way, without a proper installer which copies files into a certain known path you must always edit it yourself.

**Comment by unreal4u** [comment]
BTW, I don't know if the icon hash can change ( ping @f1multiviewer ) so if you could copy the application logo to a known location that won't change the filename that would be great!

**Reply by f1multiviewer** [reply]
It likely does, that png isn't the app icon I think, does a `.ico` or `.icns` file work?

**Reply by unreal4u** [reply]
I can confirm that an `.ico` file does work, but an `.icns` does not

**Comment by f1multiviewer** [comment]
Do you think potentially it would work better on Linux when packed as deb, Flatpak, RPM, or Snapcraft?

**Reply by unreal4u** [reply]
Difficult to say as I think that is kind of a big subject and everybody's opinion seems to differ. I'm no expert on that field though so I'm def. not the best person to ask this!

Strictly personally speaking I would like an rpm or deb package since that keeps management simple on the users side. You might have want to create a repository and such and compile and that kind of stuff, so it would add a lot if complexity on your side.

Others might like more a snap or flatpak package. On the other hand, the actual method also works quite well for me: I download the new version (I saw some dialogs in recent releases regarding an internal updater but haven't looked into that yet), extract to a folder and start the application. As long as the filenames don't change, it won't matter too much.

**Reply by f1multiviewer** [reply]
Yeah the auto updater isn't supported for Linux, unfortunately, maybe some time in the future!

**Reply by rijnhard** [reply]
Just throwing in my 2c, definitely flatpak. The fedora community and all the emerging immutable OS variants would lean towards that. It's also the one method you can use where you can guarantee the application has the dependencies you need not simply what the os has.
